src/lane_follow/src/line_detect.cpp

```diff
@@ -0,0 +1,16 @@
+/*
+ * Created By: Raad Khan
+ * Created On: April 23, 2017
+ * Description: Analyzes an image and detects lane lines, following them accordingly.
+ */
+
+#include <LineDetect.h>
+#include <ros/ros.h>
+
+int main(int argc, char* argv[]){
+    std::string node_name = "line_detect";
+    LineDetect line_detect(argc, argv, node_name);
```

Comment:
As I'm currently seeing it, `LineDetect` is a class which can detect lines - completely ROS independent. The next logical step probably would be to instantiate this class in a ros node which would then handle the subscribing and publishing of data - which would then be instantiated like you did with the `LineDetect` class here.